### PR TITLE
Fetch APIs from docs/* folder of repo

### LINKED ISSUE #51: Update API parsing logic to read individual API docs from libs
As discussed offline, we should move the current API docs from readme to their own markdown files and then update the API parsing logic accordingly. This will help keep markdown formatting consistent and expose individual class API's easier.

### PR DESCRIPTION
**Pull request by Vijar**
@redonkulus @jedireza 

Fixes #51

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by Vijar** [comment]
Before merging we need to move the Docs into the right folders in all the individual repos.

File: utils/createAPIKey.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright 2015, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+module.exports = function createKeyFromRepoPath(api) {
+    api = api || {};
+    return '/apis/' +
```
Comment (redonkulus) [inline utils/createAPIKey.js]: This shouldn't be hardcoded to `/apis/` right? Seems like you need more args to this function. Also, api should be renamed to something else (i.e. type or something).

File: utils/createAPIKey.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright 2015, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+module.exports = function createKeyFromRepoPath(api) {
```
Comment (redonkulus) [inline utils/createAPIKey.js]: Doc block

File: components/TopNav.jsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ var TopNav = React.createClass({
                     </NavLink>
                 </li>
                 <li className={cx({'pure-menu-selected': selected === 'apis'})}>
-                    <NavLink routeName="apis" navParams={{key: 'fluxible'}}>
+                    <NavLink routeName="apis" navParams={{slug: 'fluxible'}}>
```
Comment (redonkulus) [inline components/TopNav.jsx]: Does quick-start above need to use slug too?

File: configs/apis.js [hunk 4]
```diff
@@ -4,9 +4,20 @@ module.exports = [
         children: [
             {
                 label: 'Fluxible',
+                repo: 'fluxible',
+                path: 'docs/fluxible.md',
                 routeName: 'apis',
                 navParams: {
-                    key: 'fluxible'
+                    slug: 'fluxible'
```
Comment (redonkulus) [inline configs/apis.js]: In general, why rename to slug? Why not just keep it key?

Comment (Vijar) [inline configs/apis.js]: Honestly it was really confusing using key in multiple places.

[Here](https://github.com/yahoo/fluxible.io/blob/36834cc223b67827396bab83d96f7c5ed6f35d39/configs/routes.js#L47-L61) we use `:key` in the route and then create `params.key` by using the `key` from route...it threw me off for a while.

[So here](https://github.com/yahoo/fluxible.io/blob/36834cc223b67827396bab83d96f7c5ed6f35d39/configs/routes.js#L29-L46) what I did was use `slug` in the route and create `params.key` using the slug.

Also, slug is more semantic anyway for a url. This helped me keep things in the config and service more readable. I can explain more details offline.

File: components/TopNav.jsx (hunk 3 above)
Comment (Vijar) [inline components/TopNav.jsx]: no, `slug` is only for the `apis` routeName.

File: utils/createAPIKey.js (hunk 1 above)
Comment (Vijar) [inline utils/createAPIKey.js]: This is for the api keys only so it is probably okay to be hardcoded

**Comment by redonkulus** [comment]
@Vijar i updated this branch to use the `createKey` for docs. Also fixed up the tests. Lastly, I created the associated readme docs in each repo so the API links would work.

/cc @jedireza

**Comment by jedireza** [comment]
@redonkulus your changes look good

**Comment by Vijar** [comment]
Love it, :boat:

**Pull request merged**